sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -324,7 +324,9 @@ private[joins] object UnsafeHashedRelation {
  *
  * see http://java-performance.info/implementing-world-fastest-java-int-to-int-hash-map/
  */
-private[execution] final class LongToUnsafeRowMap(var mm: TaskMemoryManager, capacity: Int)
+private[execution] final class LongToUnsafeRowMap(
+    @transient val mm: TaskMemoryManager,
```

Comment:
@srowen Thanks for reviewing. I think the issue is whether the TaskMemoryManager should be serialized and used by the receiving end or not. If it should not be used by the receiving end, not by accident but by design as I believe, then it should not be serialized and sent over at all.